js-packages/@prestojs/viewmodel/src/fields/RelatedViewModelField.ts

```diff
@@ -1,11 +1,31 @@
-import { isViewModelClass, ViewModelConstructor, ViewModelInterface } from '../ViewModelFactory';
+import { isEqual } from '@prestojs/util';
+import {
+    FieldDataMappingRaw,
+    isViewModelClass,
+    ViewModelConstructor,
+    ViewModelInterface,
+} from '../ViewModelFactory';
 import Field, { FieldProps } from './Field';
+import ListField from './ListField';
+
+type RelatedViewModelValueType<T extends ViewModelConstructor<any>> = ViewModelInterface<
+    T['fields'],
+    any,
+    T['__pkFieldType'],
+    T['__pkType']
+>;
+type BaseRelatedViewModelValueType<T extends ViewModelConstructor<any>> =
+    | RelatedViewModelValueType<T>
+    | RelatedViewModelValueType<T>[];
+type RelatedViewModelParsableType<T extends ViewModelConstructor<any>> =
+    | FieldDataMappingRaw<T['fields']>
+    | FieldDataMappingRaw<T['fields']>[];
 
 /**
  * @expand-properties
  */
-type RelatedViewModelFieldProps<T extends ViewModelConstructor<any>> = FieldProps<
-    ViewModelInterface<T['fields'], any, T['__pkFieldType'], T['__pkType']>
+type RelatedViewModelFieldProps<T extends ViewModelConstructor<any>, FieldValueType> = FieldProps<
```

Comment:
As a general comment, am wondering about being more explicit with types. `T` is helpful because you immediately know that it's a generic type, but it's not always obvious what the type represents.

Does ending them with T (to distinguish from regular types) make sense?

eg in this case it would be something like
`type RelatedViewModelFieldProps<TargetViewModelT extends ViewModelConstructor<any>, FieldValueT>`

Reply:
Was trying to find some existing conventions in community, best I came across was this https://docs.microsoft.com/en-gb/archive/blogs/kcwalina/design-guidelines-update-names-of-generic-type-parameters which was referenced in a discussion about typescript. Seems as good as any 🤷‍♂️

Reply:
Using T as a prefix seems a very Microsoft thing to do (eg everything in COM is prefixed with I for Interface, ), but suffixes seem more common in JS-land (but my exposure to other typescript codebases is not very strong so I may be wrong)